packages/api/src/external/commonwell/link/shared.ts

```diff
@@ -171,5 +196,49 @@ export async function autoUpgradeNetworkLinks(
     }
 
     return validNetworkLinks;
+  } catch (error) {
+    log(`Error getting patient network links. Cause: ${errorToString(error)}`);
+    throw error;
   }
 }
+
+export function cwLinkToPatientData(cwLink: NetworkLink): PatientData {
+  const patient = cwLink.patient;
+
+  const firstName = patient?.details.name.flatMap(name => name.given).join(" ") ?? "";
+  const lastName = patient?.details.name.flatMap(name => name.family).join(" ") ?? "";
+  const dob = patient?.details.birthDate
+    ? buildDayjs(patient.details.birthDate).format(ISO_DATE)
+    : "";
+  const genderCode = patient?.details.gender.code;
+  const genderAtBirth = genderCode === "M" ? "M" : genderCode === "F" ? "F" : "U";
```

Comment:
let's put this logic in a func, and also not nest ternary operators - it makes the code difficult to read and test